### PR TITLE
Feature: implement bank/bank.cairo, token/token_utils.cairo and write tests 

### PR DESCRIPTION
**Pull request by Darlington02**
Implemented the functions in bank/bank.cairo, implemented the token_utils helper functions and wrote tests for the bank/bank.cairo contract.

# Pull Request type
- Feature

## Does this introduce a breaking change?

No

### CODE REVIEW

**Review by Sk8erboi84 — APPROVED** [review]

File: src/token/token_utils.cairo [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    fn transfer(
+        data_store: IDataStoreDispatcher,
+        token: ContractAddress,
+        receiver: ContractAddress,
+        amount: u128
+    ) {
+        if (amount.is_zero()) {
+            return ();
+        }
+        // transfer tokens to receiver and return if it suceeeds
```
Comment (Sk8erboi84) [inline src/token/token_utils.cairo:28]: Add a TODO for the gas limit please

File: src/token/token_utils.cairo [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    // amount - The amount of `token` to transfer.
+    fn transfer(
+        data_store: IDataStoreDispatcher,
+        token: ContractAddress,
+        receiver: ContractAddress,
+        amount: u128
+    ) {
+        if (amount.is_zero()) {
+            return ();
+        }
```
Comment (Sk8erboi84) [inline src/token/token_utils.cairo:27]: Dont forget to validate_receiver

File: src/token/token_utils.cairo [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+        // in case transfers to the receiver fail due to blacklisting or other reasons send the tokens to a holding address to avoid possible gaming through reverting transfers
+        let holding_address = data_store.get_address(keys::holding_address());
+        assert(!holding_address.is_zero(), 'empty_holding_address');
+        let amount_u256 = u256_from_felt252(amount.into());
+        let success1 = IERC20Dispatcher { contract_address: token }
+            .transfer(recipient: holding_address, amount: amount_u256);
+
+        // throw error if transfer fails
+        assert(success1 == true, 'token_transfer_failed');
```
Comment (Sk8erboi84) [inline src/token/token_utils.cairo:44]: add : "token_transfer_failed" in the error file and call it

**Comment by Sk8erboi84** [comment]
In the bank file you need to inherit role module (like in swap_handler)

**Review by Sk8erboi84 — CHANGES_REQUESTED** [review]
Please fix the comments so we can merge it

**Comment by Darlington02** [comment]
> In the bank file you need to inherit role module (like in swap_handler)

Sure I could, but I'm not sure I can find the use case for it in the Solidity Implementation

> Please fix the comments so we can merge it

yeap, on it...will make a push in few minutes:)

**Review by Sk8erboi84 — APPROVED** [review]

**Pull request merged**